src/util/schema_keywords.js

```diff
@@ -0,0 +1,165 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const js_utils = require('./js_utils');
+const { _: codegen} = require('ajv');
+
+/**
+ * @typedef {import('ajv').KeywordCxt} KeywordCxt
+ */
+
+const KEYWORDS = js_utils.deep_freeze({
+
+    // schema: { date: true } will match (new Date())
+    date: {
+        keyword: 'date',
+        errors: false,
+        statements: true,
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * @returns 
+         */
+        code(cxt) {
+            const val = codegen`${cxt.it.data}`;
+            cxt.gen.code(codegen`
+                if (${val} instanceof Date) {
+                } else {
+                `);
+                cxt.fail();
+                cxt.gen.code(codegen`}`);
+        }
+    },
+
+    // schema: { idate: true } will match (new Date()).getTime()
+    idate: {
+        keyword: 'idate',
+        errors: false,
+        statements: true,
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * @returns 
+         */
+        code(cxt) {
+            const val = codegen`${cxt.it.data}`;
+            // TODO: Remove accepting instanceof Date after converting all uses of
+            // idates in the database schema into the date format (currently most of them are
+            // already saved in ISO format which is not idate)
+            cxt.gen.code(codegen`
+                if (${val} instanceof Date) {
+                } else if (Number.isInteger(${val}) && !isNaN((new Date(${val})).getTime())) {
+                } else {
+                `);
+                cxt.fail();
+                cxt.gen.code(codegen`}`);
+        }
+    },
+
+    // schema: { objectid: true } will match (new mongodb.ObjectId()) or (new mongodb.ObjectId()).valueOf()
+    objectid: {
+        keyword: 'objectid',
+        errors: false,
+        statements: true,
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * @returns 
+         */
+        code(cxt) {
+            const valid = codegen`${cxt.it.validateName}`;
+            const val = codegen`${cxt.it.data}`;
+            cxt.gen.code(codegen`
+                switch (typeof ${val}) {
+                    case 'object':
+                        ${valid} = (${val} && ${val}.constructor && ${val}.constructor.name === 'ObjectID');
+                        break;
+                    case 'string':
+                        ${valid} = /^[0-9a-fA-F]{24}$/.test(${val});
+                        break;
+                    default:
+                        ${valid} = false;
+                        break;
+                }
+            `);
+        }
+    },
+
+    // schema: { binary: 64 } will match Buffer.alloc(64)
+    // schema: { binary: true } will match Buffer.alloc(<any>)
+    binary: {
+        keyword: 'binary',
+        errors: false,
+        statements: true,
```

Comment:
two options for schemaType here
```suggestion
        schemaType: ['boolean', 'number'],
```